src/lib/ares_init.c

```diff
@@ -1472,7 +1476,57 @@ static int init_by_resolv_conf(ares_channel channel)
     /* Catch the case when all the above checks fail (which happens when there
        is no network card or the cable is unplugged) */
     status = ARES_EFILE;
+#elif defined(__MVS__)
+
+  static struct __res_state *res = 0;
+  if (0 == res) {
+    int rc = res_init();
+    while (rc == -1 && h_errno == TRY_AGAIN) {
+      rc = res_init();
+    }
+    if (rc == -1) {
+      return ARES_ENOMEM;
+    }
+    res = __res();
+  }
+
+  int count4, count6;
+  __STATEEXTIPV6 *v6 = res->__res_extIPv6;
```

Comment:
These 3 variable declarations should be moved up to the top of the block for c89 compliance.